### PR TITLE
Add SDemo tests for libp2p and centralized network

### PR DESCRIPTION
**Pull request by shenkeyao**
* Adds tests for the other two types of network
* Extract common test code.

Closes https://github.com/EspressoSystems/HotShot/issues/992.

### CODE REVIEW

File: testing/tests/sequencing_tests.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            DAProposal<SequencingTestTypes>,
+            DAVote<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
+            CentralizedCommChannel<
+                SequencingTestTypes,
+                DAProposal<SequencingTestTypes>,
+                DAVote<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
+                StaticCommittee<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
+            >,
+            MemoryStorage<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
+            StaticCommittee<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
```
Comment (DieracDelta) [inline testing/tests/sequencing_tests.rs:153]: Nice! So many generics 😅

**Review by DieracDelta — DISMISSED** [review]
LGTM, nice! Couple of small nits

File: testing/tests/sequencing_tests.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct SequencingTestTypes;
+impl NodeType for SequencingTestTypes {
+    type ConsensusType = SequencingConsensus;
+    type Time = ViewNumber;
+    type BlockType = SDemoBlock;
+    type SignatureKey = JfPubKey<BLSSignatureScheme<Param381>>;
+    type VoteTokenType = StaticVoteToken<JfPubKey<BLSSignatureScheme<Param381>>>;
```
[the remark this answers is not in the document]
Comment (shenkeyao) [reply testing/tests/sequencing_tests.rs:61]: Good suggestion!

File: testing/tests/sequencing_tests.rs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+use hotshot_testing::{test_description::GeneralTestDescriptionBuilder, TestNodeImpl};
+use hotshot_types::{
+    data::{DAProposal, SequencingLeaf, ViewNumber},
+    traits::{node_implementation::NodeType, state::SequencingConsensus},
+    vote::DAVote,
+};
+use jf_primitives::signatures::BLSSignatureScheme;
+use tracing::instrument;
+
+// Test the libp2p network with sequencing consensus.
```
[the remark this answers is not in the document]
Comment (shenkeyao) [reply testing/tests/sequencing_tests.rs:24]: Oops, this is incorrect copy-pasting.

File: testing/tests/sequencing_tests.rs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+#[instrument]
+#[ignore]
+fn description_build() -> GeneralTestDescriptionBuilder {
```
[the remark this answers is not in the document]
Comment (shenkeyao) [reply testing/tests/sequencing_tests.rs:27]: Sounds good. We use a different `GeneralTestDescriptionBuilder` for stress tests, so it's probably better to add an `impl GeneralTestDescriptionBuilder` block with two constructors.

**Review by DieracDelta — APPROVED** [review]

**Pull request merged**